src/Actions/FluxAction.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace FluxErp\Actions;
+
+use FluxErp\Traits\BroadcastAction;
+use FluxErp\Traits\Makeable;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Events\NullDispatcher;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Str;
+use Spatie\Permission\Exceptions\UnauthorizedException;
+
+abstract class FluxAction
+{
+    use Makeable, BroadcastAction;
+
+    protected array $data;
+
+    protected array $rules = [];
+
+    protected mixed $result = null;
+
+    protected static Dispatcher $dispatcher;
+
+    protected static bool $isBroadcasting = true;
```

Comment:
i think this should default to false and require an active opt in in the action class